boa_ast/Cargo.toml

```diff
@@ -13,6 +13,7 @@ rust-version.workspace = true
 [features]
 serde = ["dep:serde", "boa_interner/serde", "bitflags/serde"]
 arbitrary = ["dep:arbitrary", "boa_interner/arbitrary", "num-bigint/arbitrary"]
+experimental = []
```

Comment:
In other cases, we have used more specific feature names (such as `intl`). Should we just name the feature `temporal`?

Reply:
We opted for this approach because `temporal` would be removed anyways when it reaches stage 4. Do other engines use a broad `experimental` flag for Temporal, or a specific `temporal` flag?

Reply:
Makes sense... but going forward, we might want to have features specific to some JavaScript characteristics (maybe a feature per built-in, things like that), so it might not hurt to have a `temporal` feature, and `experimental` would just depend on `temporal`. Could make it easier to support other experimental features in the future.

Reply:
That makes sense, since it allows users to reduce binary size if they need to.

Reply:
Should the feature flag update be included in this PR or a different PR?

Reply:
Can be done separately :)